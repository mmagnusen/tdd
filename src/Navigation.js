import React, { Component } from 'react';
import './App.css';

class Navigation extends Component {
    state = {
        loggedIn: false
      }

    toggleLoggedIn = () => {
        const { onClick } = this.props
        this.setState({ loggedIn: !this.state.loggedIn})
        onClick();
    }

    render() {
        const { loggedIn } = this.state
        return (
            <div>
                <h1>Navigation here</h1>
                <button id='login' onClick={this.toggleLoggedIn}>{loggedIn ? 'Logout' : 'Login'}</button>
            </div>
        );
    }
}

export default Navigation;
