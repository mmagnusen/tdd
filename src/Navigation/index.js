import React, { Component } from 'react';

class Navigation extends Component {
    state = {
        loggedIn: false
      }

    toggleLoggedIn = () => {
       
        this.setState({ loggedIn: !this.state.loggedIn})
    }

    render() {
        const { loggedIn } = this.state
        const { onClick } = this.props
        return (
            <div className='Navigation'>
                <h1>Navigation here</h1>
                <button id='login' onClick={onClick}>{loggedIn ? 'Logout' : 'Login'}</button>
            </div>
        );
    }
}

export default Navigation;
