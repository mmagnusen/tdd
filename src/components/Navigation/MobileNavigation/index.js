import React, { Component, Fragment } from 'react';
import Drawer from '@material-ui/core/Drawer';
import { Link } from '../../index';
import { LINK_TYPE } from '../../../constants';
import './MobileNavigation.scss';

class MobileNavigation extends Component {

    state = {
        mobileNavOpen: false
    }

    toggleMobileNavigation = () => {
        this.setState({
          mobileNavOpen: !this.state.mobileNavOpen
        })
    }

    getDropdownMenu = () => {
    
        const { mobileNavOpen } = this.state;
    
        return (
          <Drawer anchor="left" open={mobileNavOpen} classes={{paper: 'MobileNavigation-drawer'}} onClose={this.toggleMobileNavigation}>
            <div className='MobileNavigation-dropdown'>
              <div className='MobileNavigation-dropdownInner'>
                <Fragment>
                  <Link to="/" onClick={this.toggleMobileNavigation}><h3 className='MobileNavigation-link MobileNavigation-loginRegister'>Home</h3></Link>
                  <Link to='/courses/javascript-basics/introduction/overview' onClick={this.toggleMobileNavigation}><h3 className='MobileNavigation-link MobileNavigation-loginRegister'>Basics Course</h3></Link>
                  <Link to='/javascript-mechanics/syntax-parser' onClick={this.toggleMobileNavigation}><h3 className='MobileNavigation-link MobileNavigation-loginRegister'>Mechanics</h3></Link>
                  <Link to='/how-to/add-javascript' onClick={this.toggleMobileNavigation}><h3 className='MobileNavigation-link MobileNavigation-loginRegister'>How to guides</h3></Link>
                  <Link to='/concepts/javascript/classes' onClick={this.toggleMobileNavigation}><h3 className='MobileNavigation-link MobileNavigation-loginRegister'>Concepts</h3></Link>
                  <Link to='/jest/what-is-jest' onClick={this.toggleMobileNavigation}><h3 className='MobileNavigation-link MobileNavigation-loginRegister'>Jest</h3></Link>
                  <Link to='/resources' onClick={this.toggleMobileNavigation}><h3 className='MobileNavigation-link MobileNavigation-loginRegister'>Resources</h3></Link>
                </Fragment>
              </div>
            </div>
          </Drawer>
        )
      }

    render() {
        return (
            <div>
                <div className='MobileNavigation'>
                    <div className='MobileNavigation-inner'>
                        <Link to='/' type={LINK_TYPE.INTERNAL}>
                            <h1>Clearli</h1>
                        </Link>
                        <section className='MobileNavigation-burger' onClick={this.toggleMobileNavigation}>
                            <section className='MobileNavigation-burgerInner'>
                                <div className='MobileNavigation-burgerRow'/>
                                <div className='MobileNavigation-burgerRow'/>
                                <div className='MobileNavigation-burgerRow'/>
                            </section>
                        </section>
                    </div>
                </div>
                {this.getDropdownMenu()}
            </div>
        )
    }
};

export default MobileNavigation;
