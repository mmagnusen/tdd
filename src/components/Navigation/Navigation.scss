@import '../../constants/styles/sass-variables/breakpoints';
@import '../../constants/styles/sass-variables/colors';

$module: 'Navigation';

.#{$module} { 
    &-inner {
        height: 90px;
        display: grid;
        grid-template-columns: 200px 1fr;
        width: 80%;
        margin: 0 auto;
        @media screen and (min-width: $tabletBreakpoint) {
            width: 80%;
        }

        h1 {
            color: $turquoise;    
        }
    }

    a {
        display: flex;
        align-items: center;
    }

    &-resources {
        display: flex;
        justify-content: flex-end;
  

          a, a:visited {
              margin-left: 1rem;
              color: $ashGrey; 
          }
          a:hover {
            color: $turquoise;  
          }
    }
}