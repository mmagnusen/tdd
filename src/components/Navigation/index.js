import React, { Component } from 'react';
import { MobileNavigation, DesktopNavigation } from '../../components/';
import { Responsive} from 'semantic-ui-react';

class Navigation extends Component {

  state = {
    navigationDropDownOpen: false
  }

  toggleNavigationDropdown = () => {
    this.setState({
        navigationDropDownOpen: !this.state.navigationDropDownOpen
    })

    if (!this.state.navigationDropDownOpen) {
        document.body.style.overflow = 'hidden'
    } else {
        document.body.style.overflow = 'scroll'
    }
}


  render() {

    const { navigationDropDownOpen } = this.state;

    return (
        <nav className='Navigation'>
            <Responsive maxWidth={992}>
                <MobileNavigation 
                    toggleNavigationDropdown={this.toggleNavigationDropdown} 
                    navigationDropDownOpen={navigationDropDownOpen} />
            </Responsive>
            <Responsive minWidth={993}>
                <DesktopNavigation />
            </Responsive> 
        </nav>
    );
  }
}

export default Navigation;
