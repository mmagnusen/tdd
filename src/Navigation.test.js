import React from 'react'
import renderer from 'react-test-renderer'
import Navigation from './Navigation'
import { mount, shallow } from 'enzyme';

describe('Navigation component', () => {
    it('Renders the navigation, with react test renderer', () => {
        const navigation = renderer.create(<Navigation/>)
        expect(navigation.toJSON()).toMatchSnapshot()
    }) 
   
    it('Updates the state to loggedIn when button is clicked', () => {
        const navigation = mount(<Navigation/>);
        navigation
        .find('button#login')
        .simulate('click');
        expect(navigation.state('loggedIn')).toEqual(true);
        navigation.unmount();
    }) 
})

const fnClick = jest.fn();

describe('click event', () => {
    // it('click button should be called', () => {
    //     const component = mount(<Navigation onClick={fnClick} />);

    //     component
    //     .find('button#login')
    //     .simulate('click')
    //     expect(fnClick).toHaveBeenCalled();
    // });
});