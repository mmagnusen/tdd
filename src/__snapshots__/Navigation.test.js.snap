// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component Renders the navigation, with react test renderer 1`] = `
<div>
  <h1>
    Navigation here
  </h1>
  <button
    id="login"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
